elastic/kibana
# [Uptime] Add monitor summaries

## Summary

Resolves https://github.com/elastic/kibana/issues/38786.

Add monitor summaries. ~Code is WIP.~

Resolves #57711

## Overview
The main purpose of this PR is to allow the Uptime app's _Overview_ page to leverage a new format for Heartbeat data that we've been referring to as _States_. The purpose here is to make it easy to query how many monitors there are right now, what their status is, etc., as well as add support for grouping monitors by location. This allows us to reference documents for a single host that many heartbeat instances are creating.

### Key UI changes
- The `ErrorsList` component has been removed
- The `MonitorList` component has been refurbished to handle this new data format
  - in addition to updated back end and rendering logic, it now supports a "drawer"-style expression of the location data associated with the monitor
- The `Snapshot` and `Snapshot Histogram` components are also built using this new data format

## Testing this PR
- Bootstrap this branch and run Heartbeat from `master`
- Navigate to the Uptime page
- Ensure that you can:
  - open and close the drawers on the `MonitorList`
  - navigate from `MonitorList` to the `Monitor` page
  - the `Snapshot`'s counts correspond to the number of rows displayed in the list, as well as the counts in the accompanying bar series
  - the filtering experience is consistent with the previous version of the Uptime app
  - the integrations links work consistently with the previous version of the Uptime app

~TODO:~
~- [X] pagination~
~- [x] sorting~
~- [x] handle missing index~
~- [ ] persist expanded IDs in URL~
